modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php

```diff
@@ -171,6 +171,11 @@ protected function viewElement(FieldItemInterface $item, string $langcode): arra
     // oe_theme_helper_featured_media template.
     $cacheability->addCacheableDependency($media);
 
+    // Add the entity object and field_name to the build array.
+    $build['#entity'] = $item->getEntity();
+    $build['#field_name'] = $item->getFieldDefinition()->getName();
+    $cacheability->applyTo($build);
```

Comment:
Why do we need to apply cache meta data here?

Reply:
Antonio suggested to  move the logic we have in the formatter inside "template_preprocess_oe_theme_helper_featured_media()"